ios/browser/api/ads/brave_ads.mm

```diff
@@ -147,6 +147,7 @@ - (instancetype)initWithStateStoragePath:(NSString*)path {
     adEventCache = new brave_ads::AdEventCache();
 
     adsClient = new AdsClientIOS(self);
+    adsClientNotifier = new brave_ads::AdsClientNotifier();
```

Comment:
nit: Why are we using raw pointer allocations? You can use `unique_ptr` in Objective-C++.

Reply:
Sure. we will raise a followup issue and resolve this soon.